### ISSUE TITLE
Can't connect to AESM service on Windows with the latest Intel PSW

### ISSUE THREAD

**Issue by jxyang** [openenclave/openenclave#1810]
Steps to reproduce:

* Install the latest PSW ( v2.3.100.49777) on a SGX-enabled Windows machine
* clone and build openenclave following [instructions](https://github.com/microsoft/openenclave/blob/master/docs/GettingStartedDocs/GettingStarted.Windows.md)
* run ctest -V -R tests/asem and it will fail

I have verified the test passes for PSW v2.2.100.47975.

My investigation reveals on Windows we use COM interface to connect to the aesm service, and the CLSID is hardcoded as (in aesm.c):
```
static const CLSID CLSID_AESMInterface = {
        0x82367CAB,
        0xF2B9,
        0x461A,
        {0xB6, 0xC6, 0x88, 0x9D, 0x13, 0xEF, 0xC6, 0xCA}};
```
It seems in PSW ( v2.3.100.49777)  the `CLSID` of asem has been changed because I can't find the particular `CLSID` in the registry after installing PSW ( v2.3.100.49777).

**Comment by jxyang** [comment]
@CodeMonkeyLeet

**Comment by CodeMonkeyLeet** [comment]
Intel removed their COM interface dependencies per Microsoft recommendations for app compatibility with packaged applications. We will need to look at their RPC interface or taking a direct dependency on their enclave common library on Windows for AESM.

**Comment by jxyang** [comment]
Since Intel is moving to replace AESM with DCAP/libsgx, after discussing with @CodeMonkeyLeet, we decided to remove the test part related to creating COM object based on AESM CLSID.

**Comment by jxyang** [comment]
With PR #1865, we have made it clear that the AESM flow is not compatible with PSW 2.3 on Windows (rather, DCAP is the supported flow). Therefore I think we can close this one.

**Comment by CodeMonkeyLeet** [comment]
@jxyang #1865 documents the current state of affairs, which I do not believe is ideal given that we've not made the decision to remove AESM support entirely. Given that the PSW releases will often contain security fixes moving forwards as well, freezing the support to v2.2 is likely problematic.

We should use this to track switching the AESM build configuration to also depend on the EnclaveCommonAPI as it does in the DCAP case, with removal of oe_get_report for EPID quotes.

**Comment by parthsane** [comment]
@CodeMonkeyLeet Could you help me get more information on the RPC interface, maybe the documentation?

**Comment by CodeMonkeyLeet** [comment]
@parthsane The RPC interface remains officially undocumented by Intel, so our current direction is to simply take a dependency on the EnclaveCommon API package that comes with the Windows DCAP drop today. That comes with a debug lib so it's possible to reverse engineer the RPC interface, but given that Intel has not committed to providing that as a stable API surface (much like the original COM interface), I wouldn't bother.

**Comment by radhikaj** [comment]
@jxyang @CodeMonkeyLeet Can we close this now that we can connect to AESM with the latest PSW?

**Comment by CodeMonkeyLeet** [comment]
Fixed by #2250. OE SDK no longer depends on the AESM COM interface deprecated in PSW 2.3+.
